fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -272,14 +272,13 @@ public class LuceneRecordCursor implements BaseCursor<IndexEntry> {
     }
 
     private void ensurePrimaryKeyIsInSort() {
+        // precondition: sort is by partition key (see LucenePartitioner.isSortedByPartitionField())
+        // so, either partition field + primary key (explicitly) or just partition field.
```

Comment:
The fact that this method is so tightly coupled with `LucenePartitioner.isSortedByPartitionField()` isn't great, and it would be better if it was all contained in one place. Could `isSortedByPartitionField()` also do this logic and return a simple class with the two booleans in the `Pair`, and the updated `sort`